source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -243,33 +257,41 @@ void Filter::parseV1Header(char* buf, size_t len) {
   if (line_parts[1] != "UNKNOWN") {
     // If protocol not UNKNOWN, src and dst addresses have to be present.
     if (line_parts.size() != 6) {
-      throw EnvoyException("failed to read proxy protocol");
+      ENVOY_LOG(debug, "failed to read proxy protocol");
+      return false;
     }
 
     // TODO(gsagula): parseInternetAddressAndPort() could be modified to take two string_view
     // arguments, so we can eliminate allocation here.
-    if (line_parts[1] == "TCP4") {
-      proxy_protocol_header_.emplace(
-          WireHeader{0, Network::Address::IpVersion::v4,
-                     Network::Utility::parseInternetAddressAndPort(
-                         std::string{line_parts[2]} + ":" + std::string{line_parts[4]}),
-                     Network::Utility::parseInternetAddressAndPort(
-                         std::string{line_parts[3]} + ":" + std::string{line_parts[5]})});
-    } else if (line_parts[1] == "TCP6") {
-      proxy_protocol_header_.emplace(
-          WireHeader{0, Network::Address::IpVersion::v6,
-                     Network::Utility::parseInternetAddressAndPort(
-                         "[" + std::string{line_parts[2]} + "]:" + std::string{line_parts[4]}),
-                     Network::Utility::parseInternetAddressAndPort(
-                         "[" + std::string{line_parts[3]} + "]:" + std::string{line_parts[5]})});
-    } else {
-      throw EnvoyException("failed to read proxy protocol");
+    try {
+      if (line_parts[1] == "TCP4") {
+        proxy_protocol_header_.emplace(
+            WireHeader{0, Network::Address::IpVersion::v4,
+                       Network::Utility::parseInternetAddressAndPort(
+                           std::string{line_parts[2]} + ":" + std::string{line_parts[4]}),
+                       Network::Utility::parseInternetAddressAndPort(
+                           std::string{line_parts[3]} + ":" + std::string{line_parts[5]})});
+      } else if (line_parts[1] == "TCP6") {
+        proxy_protocol_header_.emplace(
+            WireHeader{0, Network::Address::IpVersion::v6,
+                       Network::Utility::parseInternetAddressAndPort(
+                           "[" + std::string{line_parts[2]} + "]:" + std::string{line_parts[4]}),
+                       Network::Utility::parseInternetAddressAndPort(
+                           "[" + std::string{line_parts[3]} + "]:" + std::string{line_parts[5]})});
+      } else {
+        ENVOY_LOG(debug, "failed to read proxy protocol");
+        return false;
+      }
+    } catch (const EnvoyException& ee) {
```

Comment:
Are you planning on doing a follow up to remove exceptions from this API also? I think it would be pretty easy to add a non-exception variant for use in this PR?

Reply:
Do we prefer to append `legacy` to the name of the current (exception throwing) variant, and update any usages (with the understanding that eventually every occurrence will be replaced by the new variant), or rather name the new variant something like `parseInternetAddressAndPortNoExc`?

Reply:
I would probably just do `parseInternetAddressAndPortNoExc` but I don't feel strongly about it. The other one can just be implemented in terms of the no exception one. Thank you!

/wait

Reply:
Or `parseInternetAddressAndPortNoThrow`

Reply:
Yeah ^ SGTM.